### ISSUE TITLE
Rate limit LUA ifportup checks

### ISSUE THREAD

**Issue by Someguy123** [PowerDNS/pdns#8261]
- Program: Authoritative 
 - Issue type: Bug report (and sort-of Feature request) 

### Short description

When using LUA `ifportup` - pDNS excessively checks the given IP addresses every second (sometimes more than that). 

This is problematic as for some applications (e.g. MariaDB/MySQL) it causes a lot of log spam due to the constant opening and closing of connections, not to mention the constant opening of new connections at high speed may cause false alarms with some datacenter firewalls/IDS.

With three clustered nameservers, we've found each IP in the list would be getting between 5-10 new connections every second.

There doesn't appear to be any way to control the speed at which it polls for health checks, and the TTL doesn't seem to affect this at all.

This is further excaberated by this issue: https://github.com/PowerDNS/pdns/issues/7988 - modifying the record will result in all nameservers in a cluster running health checks **even more frequently**. 

### Environment

 - Operating system: Tested on both Ubuntu 16.04 and 18.04
 - Software version: 0.0.17463.0.master.gd6f4a9fd8c (Aug 30 2019 23:13:15 - latest master) and 0.0.16227.0.gb4b9cf2d68 (Mar 28 2019 23:44:42)
 - Software source: Built from `master` using configure options: `./configure --with-modules="gmysql gpgsql geoip" --enable-libsodium --enable-systemd`
 - Backend/Clustering: NATIVE cluster of three nameservers using gmysql plugin connected to MariaDB Server 10.3

### Steps to reproduce

1. Enable LUA records for a zone
2. Add a `LUA` record using ifportup() such as `AAAA "ifportup(3306, {'2001:abc::1', '2001:abc::2', '2001:abc::3'}, {selector='random'})"` - any TTL is fine, we tested with `120` TTL.
3. PowerDNS logs (with a high log level) should show constant LUA health monitoring checks happening around 1-2 times per second. (or destination application logs should show constant connections being opened and closed by the nameserver(s)).

PowerDNS configuration:

```
local-address=REDACTED
local-ipv6=REDACTED
local-port=53
query-local-address6=REDACTED
query-local-address=REDACTED
# for better geoip, allows dns servers to send client IP
edns-subnet-processing=yes
disable-axfr=yes
master=no
slave=no
version-string=anonymous
allow-unsigned-notify=no
allow-unsigned-supermaster=no
expand-alias=true
resolver=8.8.4.4,4.2.2.3,2a07:e00::333,2607:5600:c6::333
launch=gmysql,geoip
gmysql-host = localhost
gmysql-port = 3306
gmysql-dbname = pdns
gmysql-user = pdns
gmysql-password = REDACTED
gmysql-dnssec=yes
loglevel=9
log-dns-queries=yes
default-ttl = 600

geoip-database-files=mmdb:/usr/share/GeoIP/GeoLite2-City.mmdb

default-soa-name = ns1.example.org
default-soa-edit = INCEPTION-EPOCH
default-soa-mail = hostmaster.example.org
```

### Expected behaviour

PowerDNS should only be running health checks in a reasonable amount of time relative to the TTL, e.g. for a 120 second TTL, it makes little sense to check more than every 60 seconds.

At the very least, there should be a configuration option to control the polling rate.

### Actual behaviour

PowerDNS logs show that the constant connection opening is causing false positives, with servers showing as DOWN due to timeouts, and coming back online just 1-2 seconds later, possibly due to MariaDB unable to keep up with the unusual amount of new connections.

```
    Aug 30 23:17:25 ns2 pdns_server[28493]: Lua record monitoring declaring TCP/IP [REDACTED]:3306 DOWN: timeout while connecting to [REDACTED]:3306
    ​
    Aug 30 23:17:27 ns2 pdns_server[28493]: Lua record monitoring declaring TCP/IP [REDACTED]:3306 UP!
```

Below is the error.log from a server in our MariaDB cluster, which is getting constant connections at least once per second from 2 of our nameservers (real nameserver addresses replaced with 'alpha' and 'beta'). 

```
2019-08-31  3:30:12 13329 [Warning] Aborted connection 13329 to db: 'unconnected' user: 'unauthenticated' host: 'beta' (This connection closed normally without authentication)
2019-08-31  3:30:13 13330 [Warning] Aborted connection 13330 to db: 'unconnected' user: 'unauthenticated' host: 'alpha' (This connection closed normally without authentication)
2019-08-31  3:30:13 13331 [Warning] Aborted connection 13331 to db: 'unconnected' user: 'unauthenticated' host: 'alpha' (This connection closed normally without authentication)
2019-08-31  3:30:13 13332 [Warning] Aborted connection 13332 to db: 'unconnected' user: 'unauthenticated' host: 'beta' (This connection closed normally without authentication)
2019-08-31  3:30:14 13333 [Warning] Aborted connection 13333 to db: 'unconnected' user: 'unauthenticated' host: 'alpha' (This connection closed normally without authentication)
2019-08-31  3:30:14 13334 [Warning] Aborted connection 13334 to db: 'unconnected' user: 'unauthenticated' host: 'alpha' (This connection closed normally without authentication)
2019-08-31  3:30:14 13335 [Warning] Aborted connection 13335 to db: 'unconnected' user: 'unauthenticated' host: 'beta' (This connection closed normally without authentication)
2019-08-31  3:30:15 13336 [Warning] Aborted connection 13336 to db: 'unconnected' user: 'unauthenticated' host: 'alpha' (This connection closed normally without authentication)
2019-08-31  3:30:15 13337 [Warning] Aborted connection 13337 to db: 'unconnected' user: 'unauthenticated' host: 'alpha' (This connection closed normally without authentication)
2019-08-31  3:30:15 13338 [Warning] Aborted connection 13338 to db: 'unconnected' user: 'unauthenticated' host: 'beta' (This connection closed normally without authentication)
2019-08-31  3:30:16 13339 [Warning] Aborted connection 13339 to db: 'unconnected' user: 'unauthenticated' host: 'alpha' (This connection closed normally without authentication)
2019-08-31  3:30:16 13340 [Warning] Aborted connection 13340 to db: 'unconnected' user: 'unauthenticated' host: 'alpha' (This connection closed normally without authentication)
2019-08-31  3:30:16 13341 [Warning] Aborted connection 13341 to db: 'unconnected' user: 'unauthenticated' host: 'beta' (This connection closed normally without authentication)
2019-08-31  3:30:17 13342 [Warning] Aborted connection 13342 to db: 'unconnected' user: 'unauthenticated' host: 'alpha' (This connection closed normally without authentication)
2019-08-31  3:30:17 13343 [Warning] Aborted connection 13343 to db: 'unconnected' user: 'unauthenticated' host: 'alpha' (This connection closed normally without authentication)
2019-08-31  3:30:17 13344 [Warning] Aborted connection 13344 to db: 'unconnected' user: 'unauthenticated' host: 'beta' (This connection closed normally without authentication)
```

**Comment by Cin855** [comment]
i suggest you get off my network

**Comment by mnordhoff** [comment]
Fixed by #8249?

**Comment by Habbie** [comment]
> Fixed by #8249 ?

I'd say so!
